python/nav/mibs/juniper_mib.py

```diff
@@ -146,6 +157,61 @@ def get_fru_status(self, internal_id):
     get_fan_status = get_fru_status
     get_power_supply_status = get_fru_status
 
+    @defer.inlineCallbacks
+    def get_all_sensors(self):
+        """Returns a Deferred whose result is a list of sensor dictionaries"""
+        result = []
+        for table, config in SENSOR_TABLES.items():
+            sensors = yield self._get_sensors(config)
+            result.extend(sensors)
+        defer.returnValue(result)
+
+    @defer.inlineCallbacks
+    def _get_sensors(self, config):
+        """
+        Collects sensor columns according to the config dict, and translates
+        the results into sensor dicts.
+
+        """
+        columns = [config['descr'], config['readout']]
+
+        result = (
+            yield self.retrieve_columns(columns)
+            .addCallback(self.translate_result)
+            .addCallback(reduce_index)
+        )
+
+        sensors = (
+            self._row_to_sensor(config, index, row) for index, row in result.items()
+        )
+
+        defer.returnValue([s for s in sensors if s])
+
+    def _row_to_sensor(self, config, index, row):
+        """
+        Converts a collect SNMP table row into a sensor dict, using the
+        options defined in the config dict.
+
+        """
+        internal_name = config['internal_prefix'] + str(index)
+        descr = row.get(config['descr'], internal_name) + " Temperature"
+
+        # Dont include sensor if temperature not set
+        readout = row.get(config['readout'], 0)
+        if not readout:
+            return
+
+        return {
+            'oid': str(self.nodes[config['readout']].oid + index),
+            'unit_of_measurement': config['unit'],
+            'precision': 0,
+            'scale': None,
+            'description': descr,
+            'name': descr,
+            'internal_name': internal_name,
+            'mib': 'SPAGENT-MIB',
```

Comment:
The MIB is definitely **not** `SPAGENT-MIB` in this case :-)

```suggestion
            'mib': self.get_module_name(),
```

Reply:
fixed 🙏